### ISSUE TITLE
deprecationerrors query hangs for 10-20 seconds

### ISSUE THREAD

**Issue by drifteaur** [craftcms/cms#2539]
### Description

I'm working on a Craft 3 site, and page loads have gotten very slow all of a sudden. Using the Yii debug toolbar, I was able to find that it was due to this query, running it on a MySQL console takes about 17 seconds on my machine? It fetches some metadata about the _deprecationerrors_ table I guess.

```
SELECT DISTINCT
    `kcu`.`CONSTRAINT_NAME` AS `name`,
    `kcu`.`COLUMN_NAME` AS `column_name`,
    `tc`.`CONSTRAINT_TYPE` AS `type`,
    CASE
        WHEN NULL IS NULL AND `kcu`.`REFERENCED_TABLE_SCHEMA` = `sch`.`name` THEN NULL
        ELSE `kcu`.`REFERENCED_TABLE_SCHEMA`
    END AS `foreign_table_schema`,
    `kcu`.`REFERENCED_TABLE_NAME` AS `foreign_table_name`,
    `kcu`.`REFERENCED_COLUMN_NAME` AS `foreign_column_name`,
    `rc`.`UPDATE_RULE` AS `on_update`,
    `rc`.`DELETE_RULE` AS `on_delete`,
    `kcu`.`ORDINAL_POSITION` as `position`
FROM (SELECT DATABASE() AS `name`) AS `sch`
INNER JOIN `information_schema`.`KEY_COLUMN_USAGE` AS `kcu`
    ON `kcu`.`TABLE_SCHEMA` = COALESCE(NULL, `sch`.`name`) AND `kcu`.`CONSTRAINT_SCHEMA` = `kcu`.`TABLE_SCHEMA` AND `kcu`.`TABLE_NAME` = 'deprecationerrors'
LEFT JOIN `information_schema`.`REFERENTIAL_CONSTRAINTS` AS `rc`
    ON `rc`.`CONSTRAINT_SCHEMA` = `kcu`.`TABLE_SCHEMA` AND `rc`.`CONSTRAINT_NAME` = `kcu`.`CONSTRAINT_NAME`
LEFT JOIN `information_schema`.`TABLE_CONSTRAINTS` AS `tc`
    ON `tc`.`TABLE_SCHEMA` = `kcu`.`TABLE_SCHEMA` AND `tc`.`CONSTRAINT_NAME` = `kcu`.`CONSTRAINT_NAME`
ORDER BY `kcu`.`ORDINAL_POSITION` ASC
```

### Steps to reproduce

1. No idea, one day the site was very fast, next day page loads are taking very long.

### Additional info

- Craft version: 3.0.0-RC13
- PHP version: 7.0.24
- Database driver & version: MySQL 5.5.5
- Plugins & versions:

**Comment by angrybrad** [comment]
Should already be fixed in https://github.com/craftcms/cms/commit/a2609d49d6eccd2e0457f3470188795382ad3ef1 and will be included in today's release.

**Comment by bekeeg** [comment]
@angrybrad I've recently updated my local to 3.0.13.2 and the commit you referenced doesn't appear to have been addressed. That line in the file is still:

 'enableSchemaCache' => !YII_DEBUG,

**Comment by brandonkelly** [comment]
@bekeeg `true` was swapped with `!YII_DEBUG` later on.

**Comment by bekeeg** [comment]
@brandonkelly Thanks for the quick response!

I don't mean to hijack this closed thread, but I'm having the same issue as the OP and am having trouble tracking down the culprit. Any insight would be appreciated!

**Comment by brandonkelly** [comment]
@bekeeg The original issue was due to a Yii issue (yiisoft/yii2#15776), which was fixed in Yii 2.0.14.2, so 3.0.0-RC15 we started requiring that version, and stopped forcing the schema cache to be enabled when Dev Mode is enabled, as it could likely lead to other unexpected behavior when the schema is potentially changing often (9498b14bb4e53cbcef82971d6cd3845b25aa34ce).

Make sure you’re running  Yii 2.0.14.2+ (Craft 3.0.13.2 requires Yii ~2.0.15.1 so that should be the case, but worth checking). If still having issues, try installing Craft on a new database connection that doesn't have a lot of other databases and see if that helps – the original issue turned out to only occur when there were lots of other DBs on the server.
